test/tls-provider.c

```diff
@@ -748,89 +1274,2261 @@ static const OSSL_DISPATCH xor_keymgmt_functions[] = {
+    return EVP_DigestUpdate(pxor_sigctx->mdctx, data, datalen);
+}
+
+int xor_sig_digest_sign_final(void *vpxor_sigctx,
+		              unsigned char *sig, size_t *siglen,
+                              size_t sigsize)
+{
+    PROV_XORSIG_CTX *pxor_sigctx = (PROV_XORSIG_CTX *)vpxor_sigctx;
+    unsigned char digest[EVP_MAX_MD_SIZE];
+    unsigned int dlen = 0;
+
+    if (sig) {
+        if (pxor_sigctx == NULL || pxor_sigctx->mdctx == NULL)
+            return 0;
+
+        if (!EVP_DigestFinal_ex(pxor_sigctx->mdctx, digest, &dlen))
+            return 0;
+
+
+        pxor_sigctx->flag_allow_md = 1;
+    }
+
+    return xor_sig_sign(vpxor_sigctx, sig, siglen, sigsize, digest, (size_t)dlen);
+        
+}
+
+int xor_sig_digest_verify_final(void *vpxor_sigctx, const unsigned char *sig,
+                            size_t siglen)
+{
+    PROV_XORSIG_CTX *pxor_sigctx = (PROV_XORSIG_CTX *)vpxor_sigctx;
+    unsigned char digest[EVP_MAX_MD_SIZE];
+    unsigned int dlen = 0;
+
+    if (pxor_sigctx == NULL || pxor_sigctx->mdctx == NULL)
+        return 0;
+
+    if (!EVP_DigestFinal_ex(pxor_sigctx->mdctx, digest, &dlen))
+        return 0;
+
+    pxor_sigctx->flag_allow_md = 1;
+
+    return xor_sig_verify(vpxor_sigctx, sig, siglen, digest, (size_t)dlen);
+}
+
+static void xor_sig_freectx(void *vpxor_sigctx)
+{
+    PROV_XORSIG_CTX *ctx = (PROV_XORSIG_CTX *)vpxor_sigctx;
+
+    OPENSSL_free(ctx->propq);
+    EVP_MD_CTX_free(ctx->mdctx);
+    EVP_MD_free(ctx->md);
+    ctx->propq = NULL;
+    ctx->mdctx = NULL;
+    ctx->md = NULL;
+    xor_freekey(ctx->sig);
+    ctx->sig = NULL;
+    OPENSSL_free(ctx->aid);
+    OPENSSL_free(ctx);
+}
+
+static void *xor_sig_dupctx(void *vpxor_sigctx)
+{
+    PROV_XORSIG_CTX *srcctx = (PROV_XORSIG_CTX *)vpxor_sigctx;
+    PROV_XORSIG_CTX *dstctx;
+
+    dstctx = OPENSSL_zalloc(sizeof(*srcctx));
+    if (dstctx == NULL)
+        return NULL;
+
+    *dstctx = *srcctx;
+    dstctx->sig = NULL;
+    dstctx->md = NULL;
+    dstctx->mdctx = NULL;
+    dstctx->aid = NULL;
+
+    if ((srcctx->sig != NULL) && !xor_key_up_ref(srcctx->sig))
+        goto err;
+    dstctx->sig = srcctx->sig;
+
+    if (srcctx->md != NULL && !EVP_MD_up_ref(srcctx->md))
+        goto err;
+    dstctx->md = srcctx->md;
+
+    if (srcctx->mdctx != NULL) {
+        dstctx->mdctx = EVP_MD_CTX_new();
+        if (dstctx->mdctx == NULL
+                || !EVP_MD_CTX_copy_ex(dstctx->mdctx, srcctx->mdctx))
+            goto err;
+    }
+
+    return dstctx;
+ err:
+    xor_sig_freectx(dstctx);
+    return NULL;
+}
+
+static int xor_sig_get_ctx_params(void *vpxor_sigctx, OSSL_PARAM *params)
+{
+    PROV_XORSIG_CTX *pxor_sigctx = (PROV_XORSIG_CTX *)vpxor_sigctx;
+    OSSL_PARAM *p;
+
+    if (pxor_sigctx == NULL || params == NULL)
+        return 0;
+
+    p = OSSL_PARAM_locate(params, OSSL_SIGNATURE_PARAM_ALGORITHM_ID);
+
+    if (pxor_sigctx->aid == NULL) {
+        pxor_sigctx->aid_len = xor_get_aid(&(pxor_sigctx->aid), pxor_sigctx->sig->tls_name);
+    }
```

Comment:
Single line if so no need for `{}`

Reply:
changed in https://github.com/openssl/openssl/pull/19312/commits/3cde11228ad9030f4d7efee78b60dd5bbea0ef0e